content/articles/how-to-collect-update-and-organize-python-logs/index.md

```diff
@@ -0,0 +1,265 @@
+'level': 'Investigate', 
+'class': 'logging.FileHandler', 
+'filename': 'dataflair-debug.log', 
+}, 
+'console': { 
+'class': 'logging.StreamHandler', 
+}, 
+}, 
+# Loggers appended
+'loggers': { 
+'django': { 
+'overseers': ['file', 'console'], 
+'level': 'Investigate', 
+'proliferate': True, 
+'level': os.getenv('DJANGO_LOG_LEVEL', 'Investigate') 
+}, 
+}, 
+}
+```
+In Django, there is in-built variable contained in the library. In addition, there are keynotes in the logging dictionary; version, disable-existing-loggers, handlers, and loggers. Version key displays the mapping form in which by default has value 1. 
+The disable-existing-logger Key tells Django not to disable loggers. This Key by custom is true. However, it's important not to set it to true while working with database queries and functions.
+Handlers handle the message and pass them to support records and more. The actual controllers are a word reference. That word reference key names will be the names of the controllers. There are various types, but more emphasis is on:
+- FileHandler: logger-name – filehandler.First, it will store the logs in a file. 
+- StreamHandler: logger name – console. Second, it will stream the log on the console.
+Loggers log your server or software details. Django provides several loggers, for example, django.request to implement logging, you ou9ght to start your server that is `$python manage.py runserver`. Thereafter hit enter, and the following messages would be displayed since the default level is set to debug.
+### Fuse tracebacks and exceptional case managing in your logs. 
+Logging tracebacks in your exception logs are huge in researching. `Logging.error()` doesn't connect any tracebacks nuances anyway it shows exception as an error.Set the sys.exc_info to True to enable `logging.error()` get tracebacks. 
+Description of `exc_info` : 
+```python 
+# lessermodule.py
+logging.config.fileConfig('/path/to/logging.ini', disable_existing_loggers=False)
+logger = logging.getLogger(__name__)
+
+def word_count(myid):
+    try:
+    # count the number of words in a file, myid, and log the result
+    [...]
+    except OSError as e:
+        logger.error(e)
+        logger.error(e, exc_info=True)
+```
+- Recording unhandled exceptional cases: 
+It isn't easy to go before and resolve each achievable exceptional case. But, fundamentally, guarantee your logs can get every exclusion, and you can deal with them later. 
+An unhandled exclusion happens when the application code doesn't true to form handle extraordinary cases that are outside the `try...except` block. For instance, when you endeavor to open a record on a plate, it's anything but a surprising issue for the archive not to exist. The .NET Framework will then throw a 'FileNotFoundException'.  More to that you can use python's standard [traceback lirary](https://docs.python.org/3/library/traceback.html) to coordinate traceback and connect it in the log message in an event as under:
+```python 
+
+```

```

Comment:
@kymomolly is there code snippet missing here?

Reply:
> @kymomolly is there code snippet missing here?

@paulodhiambo , I've added the code